spec/springboard/client/uri_spec.rb

```diff
@@ -12,10 +12,35 @@
   end
 
   describe "parse" do
-    it "should return a URI based on the given string" do
-      uri = described_class.parse('/some_path')
-      expect(uri).to be_a(described_class)
-      expect(uri.to_s).to eq('/some_path')
+    describe "when called with a URI object" do
+      it "should return a cloned URI object" do
+        parsed_uri = described_class.parse(uri)
+        expect(uri.class).to eq(parsed_uri.class)
+        expect(uri.object_id).not_to eq(parsed_uri.object_id)
+      end
+    end
+
+    describe "when called with a URI string" do
+      it "should return a URI based on the given string" do
+        uri = described_class.parse('/some_path')
+        expect(uri).to be_a(described_class)
+        expect(uri.to_s).to eq('/some_path')
+      end
+    end
+  end
+
+  describe "dup" do
+    it "should return a duplicate URI object" do
+      dup_uri = uri.dup
+      expect(uri.class).to eq(dup_uri.class)
+      expect(uri.object_id).not_to eq(dup_uri.object_id)
```

Comment:
Probably good to assert that the new one has the same `to_s` representation here, too.

Reply:
Makes sense. Added